Xamarin.Forms.Platform.Android/FastRenderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,488 @@
+
+			if (oldElement != null)
+			{
+				oldElement.PropertyChanged -= OnElementPropertyChanged;
+			}
+
+			Color currentColor = oldElement?.BackgroundColor ?? Color.Default;
+			if (element.BackgroundColor != currentColor)
+			{
+				UpdateBackgroundColor();
+			}
+
+			MediaElement.PropertyChanged += OnElementPropertyChanged;
+			MediaElement.SeekRequested += SeekRequested;
+			MediaElement.StateRequested += StateRequested;
+
+			if (_tracker == null)
+			{
+				// Can't set up the tracker in the constructor because it access the Element (for now)
+				SetTracker(new VisualElementTracker(this));
+			}
+
+			OnElementChanged(new ElementChangedEventArgs<MediaElement>(oldElement as MediaElement, MediaElement));
+
+			EffectUtilities.RegisterEffectControlProvider(this, oldElement, element);
+
+			Performance.Stop(reference);
+		}
+
+		void StateRequested(object sender, StateRequested e)
+		{
+			switch (e.State)
+			{
+				case MediaElementState.Playing:
+					_view.Start();
+					Controller.CurrentState = _view.IsPlaying ? MediaElementState.Playing : MediaElementState.Stopped;
+					break;
+
+				case MediaElementState.Paused:
+					if (_view.CanPause())
+					{
+						_view.Pause();
+						Controller.CurrentState = MediaElementState.Paused;
+					}
+					break;
+
+				case MediaElementState.Stopped:
+					_view.Pause();
+					_view.SeekTo(0);
+
+					Controller.CurrentState = _view.IsPlaying ? MediaElementState.Playing : MediaElementState.Stopped;
+					break;
+			}
+
+			UpdateLayoutParameters();
+			Controller.Position = _view.Position;
+		}
+
+		void SeekRequested(object sender, SeekRequested e)
+		{
+			Controller.Position = _view.Position;
+		}
+
+		void IVisualElementRenderer.SetLabelFor(int? id)
+		{
+			if (_defaultLabelFor == null)
+			{
+				_defaultLabelFor = LabelFor;
+			}
+
+			LabelFor = (int)(id ?? _defaultLabelFor);
+		}
+		void SetTracker(VisualElementTracker tracker)
+		{
+			_tracker = tracker;
+		}
+
+		void UpdateBackgroundColor()
+		{
+			SetBackgroundColor(Element.BackgroundColor.ToAndroid());
+		}
+
+		void IVisualElementRenderer.UpdateLayout()
+		{
+			var lp = _view.LayoutParameters;
+			lp.Width = Width;
+			lp.Height = Height;
+			_view.LayoutParameters = lp;
+
+			_tracker?.UpdateLayout();
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (_isDisposed)
+			{
+				return;
+			}
+
+			_isDisposed = true;
+
+			if (disposing)
+			{
+				SetOnClickListener(null);
+				SetOnTouchListener(null);
+
+				_automationPropertiesProvider?.Dispose();
+				_tracker?.Dispose();
+
+				if (Element != null)
+				{
+					Element.PropertyChanged -= OnElementPropertyChanged;
+
+					if (Platform.GetRenderer(Element) == this)
+						Element.ClearValue(Platform.RendererProperty);
+				}
+			}
+
+			base.Dispose(disposing);
+		}
+
+		void OnElementChanged(ElementChangedEventArgs<MediaElement> e)
+		{
+			if (e.OldElement != null)
+			{
+				ReleaseControl();
+			}
+
+			if (e.NewElement != null && !_isDisposed)
```

Comment:
we shouldn't check _isDisposed we should check maybe if Control (_view)  is null